lib/rules/no-global-setup.js

```diff
@@ -10,24 +10,37 @@ var setupRegexp = /^(before|after)(Each|All)$/
 
 module.exports = function (context) {
   var suiteDepth = 0
+  var hasGlobalSetup = false
+  var hasRootSuite = false
+  var globalSetupNames = []
   return {
     CallExpression: function (node) {
       if (suiteRegexp.test(node.callee.name)) {
+        hasRootSuite = true
         suiteDepth++
       } else if (setupRegexp.test(node.callee.name) && suiteDepth === 0) {
-        context.report({
-          data: {
-            name: node.callee.name
-          },
-          message: 'Do not use `{{name}}` outside a `describe`.',
-          node
-        })
+        hasGlobalSetup = true
+        globalSetupNames.push(node.callee.name)
       }
     },
     'CallExpression:exit': function (node) {
       if (suiteRegexp.test(node.callee.name)) {
         suiteDepth--
       }
+    },
+    'Program:exit': function () {
+      if (hasGlobalSetup && hasRootSuite) {
+        context.report({
+          data: {
+            names: globalSetupNames.join(', ')
+          },
+          message: 'Do not use `{{names}}` outside a `describe` except for in global helpers.',
+          loc: {
```

Comment:
I see what you mean, generally eslint rule return one error per node. eslint handles the case when program has multiple nodes that violate the rule. So instead of collecting `globalSetupNames` and reporting them all on the first line, would it be possible to report them once you encounter  a `globalSetupName`?